### PR TITLE
Event driven

### PR DESCRIPTION
**Pull request by Triasmus**
This makes the buy and sell blocks event driven, so that we attempt to buy immediately after hearing the announcement, and then sell once we're done buying.

I also made the config values global, since I figured that it makes sense for them to be global and I figured it would be the easiest way to get those values to the buy block.

~~I decided to leave the sell block in the main thread. It has to keep a sleeper anyway, since it does api requests to learn the current sale price.~~

This is based on ~~#62~~ master

~~The only way I've tested the Threading.Event.Wait() func is through the ctrl-c functionality. I'm confident it will work, based on the `buy_ready event triggered` log message coming when I expect after hitting ctrl-c, but it has not been tested with a real announcement.~~
It still hasn't been tested with a real announcement, but it works with a fake announcement.

### CODE REVIEW

**Comment by Triasmus** [comment]
I suggest review the changes with just the last two commits: https://github.com/CyberPunkMetalHead/gateio-crypto-trading-bot-binance-announcements-new-coins/pull/63/files/489a9241d861476b9c22ffeead43a9467ee77429..a4d65988dca02f4e5d5cb1f0394a2b18f4f25a94

File: main.py [hunk 1]
```diff
@@ -52,24 +52,199 @@
...
 
+def buy():
+    while not globals.stop_threads:
+        logger.debug('Waiting for buy_ready event')
+        globals.buy_ready.wait()
+        logger.debug('buy_ready event triggered')
+        globals.buy_ready.clear()
+        if globals.stop_threads:
+            break
+        # the buy block and logic pass
```
Comment (Triasmus) [inline main.py:63]: This line and on are straight copy/paste from main (except for and `globals.` before the now-global variables)

**Comment by Triasmus** [comment]
I'll update this pull once #59 and #62 get merged into master.

**Comment by Triasmus** [comment]
I've determined that it's not ready, even if you wanted to merge it before #59 for some reason.

I've realized that the buy block expects to have more chances to fulfill the whole order, so I shouldn't clear the Event until it's not going to want to buy any more.

**Comment by VuzzyM** [comment]
@Triasmus Today was announcements, did you test the functionality together with pr 62?

**Comment by Triasmus** [comment]
@VuzzyM I did. My internet went out for a bit, which caused some issues (as noted in discord).
I restarted the script about 10 minutes after ENS was announced. The script then immediately caught ENS as a new announcement...
Besides that, it seems like it's working correctly.

There is a potential bug I thought of yesterday morning that I need to look at before I un-draft this PR

**Comment by VuzzyM** [comment]
> @VuzzyM I did. My internet went out for a bit, which caused some issues (as noted in discord). I restarted the script about 10 minutes after ENS was announced. The script then immediately caught ENS as a new announcement... Besides that, it seems like it's working correctly.
> 
> There is a potential bug I thought of yesterday morning that I need to look at before I un-draft this PR

Okey, we are waiting for the solution

**Comment by Triasmus** [comment]
I'm not planning on doing any updates for this pull until #59 goes in, unless I get told that 59 isn't going to be going in any time soon.

**Comment by VuzzyM** [comment]
@cryptmjt ?

**Comment by Triasmus** [comment]
I believe this will work now.

After doing all the work, I made a separate branch and just copy/pasted the buy and sell blocks to their own functions in one commit (https://github.com/CyberPunkMetalHead/gateio-crypto-trading-bot-binance-announcements-new-coins/pull/63/commits/f4791022ba6e0d50369c431cbe38fd130476d840) and then made a separate commit with all the changes I actually made (https://github.com/CyberPunkMetalHead/gateio-crypto-trading-bot-binance-announcements-new-coins/pull/63/commits/c9451393dc52ccac2bc5a39b44cc62e8af015abb).

I suggest reviewing https://github.com/CyberPunkMetalHead/gateio-crypto-trading-bot-binance-announcements-new-coins/pull/63/commits/c9451393dc52ccac2bc5a39b44cc62e8af015abb with whitespace turned off if you want to see the changes I actually made.

File: globals.py [hunk 2]
```diff
@@ -1,2 +1,16 @@
+import threading
+
+buy_ready = threading.Event()
+sell_ready = threading.Event()
 stop_threads = False
 old_coins = {}
+
+#TRADE_OPTIONS config values
```
Comment (CyberPunkMetalHead) [inline globals.py:8]: Is there a particular reason behind moving the config options from yaml file to a python one?

File: main.py [hunk 3]
```diff
@@ -54,371 +54,400 @@
...
+                announcement_coin not in globals.old_coins:
+            logger.info(f'New announcement detected: {announcement_coin}')
+            if not supported_currencies:
+                supported_currencies = get_all_currencies(single=True)
+            if supported_currencies:
+                if announcement_coin in supported_currencies:
+                    logger.debug("Starting get_last_price")
+                    
+                    # get latest price object
+                    obj = get_last_price(announcement_coin, globals.pairing, False)
```
Comment (CyberPunkMetalHead) [inline main.py:84]: can we call this price for readability

Comment (Triasmus) [reply main.py:84]: `obj`? The very next line does `price = obj.last`

This also isn't one of the things I did. If you look at this commit https://github.com/CyberPunkMetalHead/gateio-crypto-trading-bot-binance-announcements-new-coins/pull/63/commits/c9451393dc52ccac2bc5a39b44cc62e8af015abb it has the changes I actually made to the buy and sell blocks.

**Pull request merged**